### ISSUE TITLE
Errors for default syntax with enabled strict checks

### ISSUE THREAD

**Issue by asyncink** [simonmichael/hledger#2119]
hi

1. when enabled strict check for payees, hledger emits errors for all transactions without payee:

    ```
    2023-11-30
      Account1  1 USD
      Account2
    ```

    ```
    Strict payee checking is enabled, and
    payee "" has not been declared.
    Consider adding a payee directive. Examples:

    payee
    ```

    this can not be fixed because there is no way to create empty tag

2. when enabled strict check for tags, hledger emits errors for accounts with implicit type:

    ```
    account Assets:Cash  ; type: C
    ```

    ```
    Strict tag checking is enabled, and
    tag "type" has not been declared.
    Consider adding a tag directive. Examples:

    tag type
    ```

    this can be fixed by `type` tag, but this feels more like temp workaround

p.s. thx for hledger!

**Comment by asyncink** [comment]
```
hledger --version
hledger 1.31, mac-x86_64
```

**Comment by asyncink** [comment]
thank you for fast reply! hledger is already cool, but with the opportunity to discuss its development and remaining errors - it’s amazing.

1. actually I don’t think that allowing `payee ""` directive is good. according to documentation, user can optionally divide transaction description into two parts with char `|`: on the left side payee and on the right side and till the end of the line — description. for me it feels consistent that omitting char `|` is the same that just leaving right side (description):

```
2023-10-30 Whole Foods | Groceries for dinner ; <~ it’s payee and description
  Account1 1 USD
  Account 2

2023-10-30 Cinema with friends ; <~ it’s just a description (since to payee was provided)
  Account1 10 USD
  Account 2

2023-10-30 ; <~ it’s just no payee and no decsription
  Account1 10 USD
  Account 2
```

so for my opinion the expected behavior is if strict payees check is fired only for transactions with payee provided (with char ‘|’) and is not firing if no description provided or provided without char ‘|’.

otherwise (current behavior) enabling payees strict check perses three transactions above as 3 different payees, but `Cinema with friends` is just a description. it can be solved like this:

```
2023-10-30 | Cinema with friends ; <~ it’s just a description (since to payee was provided)
  Account1 10 USD
  Account 2
```

but then payees check will throw error for empty payee. and also users with journals will have problems with all their already existing transactions.

**to summarize**: in my opinion, payees strict check should fire only if character `|` is provided AND only of left side part (substring leftside from the `|`) is not empty.

**Comment by simonmichael** [comment]
In principle I think payee is "Whole Foods", "", and "" respectively in those three cases, and it should be checked consistently in each case. Because it should be possible to disallow writing an entry without one of the valid payees (as with accounts, commodities, etc.)  Possibly some docs/functionality might need to be adjusted here.

**Comment by simonmichael** [comment]
I have added `payee ""` support.

Re automatically declaring the built-in tag names: `type` is the one you mentioned; `t` is another one I can think of (generated by timedot's letters notation). 

Should it be an error if the user tries to declare these tags themselves with `tag type` or `tag t` ?

**Comment by simonmichael** [comment]
The built-in tags are now auto-declared, and it's not an error if user accidentally declares them explicitly. I think this is resolved.
